ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+                data.r1.tels_with_data = tels_with_data
+                data.dl0.tels_with_data = tels_with_data
+                data.trig.tels_with_trigger = tels_with_data
+                data.inst.subarray = magic_subarray
+                
+                yield data
+                counter += 1
+        
+        return
+
+    def _build_subarray_info(self, file):
+        """
+        constructs a SubarrayDescription object from the info in an
+        EventIO/HESSSIO file
+
+        Parameters
+        ----------
+        file: HessioFile
+            The open pyhessio file
+
+        Returns
+        -------
+        SubarrayDescription :
+            instrumental information
+        """
+        telescope_ids = list(file.get_telescope_ids())
+        subarray = SubarrayDescription("MonteCarloArray")
+
+        for tel_id in telescope_ids:
+            try:
+
+                pix_pos = file.get_pixel_position(tel_id) * u.m
+                foclen = file.get_optical_foclen(tel_id) * u.m
+                mirror_area = file.get_mirror_area(tel_id) * u.m ** 2
+                num_tiles = file.get_mirror_number(tel_id)
+                tel_pos = file.get_telescope_position(tel_id) * u.m
+
+                tel = TelescopeDescription.guess(*pix_pos,
+                                                 equivalent_focal_length=foclen)
+                tel.optics.mirror_area = mirror_area
+                tel.optics.num_mirror_tiles = num_tiles
+                subarray.tels[tel_id] = tel
+                subarray.positions[tel_id] = tel_pos
+
+            except self.pyhessio.HessioGeneralError:
+                pass
+
+        return subarray
+
+
+class MAGICEventSourceROOT(EventSource):
+    """
+    EventSource for MAGIC calibrated data.
+
+    This class operates with the MAGIC data run-wise. This means that the files
+    corresponding to the same data run are loaded and processed together.
+    """
+    _count = 0
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool: ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs: dict
+            Additional parameters to be passed.
+            NOTE: The file mask of the data to read can be passed with
+            the 'input_url' parameter.
+        """
+
+        file_list = glob.glob(kwargs['input_url'])
+        file_list.sort()
+
+        # EventSource can not handle file wild cards as input_url
+        # To overcome this we substitute the input_url with first file matching
+        # the specified file mask.
```

Comment:
This is  a general issue with `EventSource` right? If so, well spotted. Maybe this issue should be fixed in EventSource then.

Reply:
We didn't want to touch the EventSource as this could have affected others. Maybe this can be fixed with the next patch?